scipy/spatial/transform/rotation.pyx

```diff
@@ -1012,14 +1015,110 @@ cdef class Rotation(object):
         return cls(quat[0] if is_single else quat, normalize=False, copy=False)
 
+    @classmethod
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def from_mrp(cls, mrp):
+        """Initialize from Modified Rodrigues Parameters (MRPs).
+
+        MRPs are a 3 element attitude description commonly used in the aerospace
+        industry.
+
+        Parameters
+        ----------
+        mrp : array_like, shape (N, 3) or (3,)
+            A single vector or a stack of vectors, where `mrp[i]` gives
+            the ith set of MRPs.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotations represented by input rotation
+            vectors.
+
+        References
+        ----------
+        .. [1] Shuster, M. D. "A Survery of Attitude Representations",
+               The Journal of Astronautical Sciences, Vol. 41, No.4, 1993,
+               pp. 475-476
+
+        Notes
+        -----
+
+        .. versionadded:: 1.5.3
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Initialize a single rotation:
+
+        >>> r = R.from_mrp([0, 0, 0.41421356])
+        >>> r.as_rotvec()
+        array([0.        , 0.        , 1.57079633])
+        >>> r.as_rotvec().shape
+        (3,)
+
+        Initialize multiple rotations in one object:
+
+        >>> r = R.from_mrp([
+        ... [0, 0, 0.41421356],
+        ... [0.41421356, 0, 0]])
+        >>> r.as_rotvec()
+        array([[0.        , 0.        , 1.57079633],
+               [1.57079633, 0.        , 0.        ]])
+        >>> r.as_rotvec().shape
+        (2, 3)
+
+        It is also possible to have a stack of a single rotaton:
+
+        >>> r = R.from_mrp([[0, 0, np.pi/2]])
+        >>> r.as_rotvec().shape
+        (1, 3)
+
+        """
+        is_single = False
+        mrp = np.asarray(mrp, dtype=float)
+
+        if mrp.ndim not in [1, 2] or mrp.shape[len(mrp.shape) - 1] != 3:
+            raise ValueError("Expected `mrp` to have shape (3,) "
+                             "or (N, 3), got {}".format(mrp.shape))
+
+        # If a single vector is given, convert it to a 2D 1 x 3 matrix but
+        # set self._single to True so that we can return appropriate objects
+        # in the `as_...` methods
+        cdef double[:, :] cmrp
+        if mrp.shape == (3,):
+            cmrp = mrp[None, :]
+            is_single = True
+        else:
+            cmrp = mrp
+
+        cdef Py_ssize_t num_rotations = cmrp.shape[0]
+        cdef double[:, :] quat = _empty2(num_rotations, 4)
+        cdef double ss, qnorm
+
+        for ind in range(num_rotations):
+            ss = _dot3(cmrp[ind, :], cmrp[ind, :])
+
+            quat[ind, 0] = 2 * cmrp[ind, 0] / (1 + ss)
+            quat[ind, 1] = 2 * cmrp[ind, 1] / (1 + ss)
+            quat[ind, 2] = 2 * cmrp[ind, 2] / (1 + ss)
+            quat[ind, 3] = (1 - ss) / (1 + ss)
```

Comment:
minor: not sure if the compiler will optimize the repeated `(1 + ss)` calls or if assigning the value to i.e., a variable once might help

Reply:
I'm not sure either, I went ahead and changed `ss` to `1+ss`